src/components/Row/Row.tsx

```diff
@@ -1,65 +1,83 @@
 import { Platform, StyleProp, ViewStyle } from 'react-native';
 import React, { createContext } from 'react';
-import { View, ViewProps } from '@bluebase/components';
-import { ScreenSizeConsumer } from '../ScreenSize';
-import { Theme } from '@bluebase/core';
-import { isHidden } from '../../helpers';
+import { SCREEN_SIZE, View, ViewProps } from '@bluebase/components';
+import { Theme, useStyles } from '@bluebase/core';
 
-const initialSize = 12;
+import { getScreenSizeFromWidth } from '../../helpers';
+import { useLayout } from '@react-native-community/hooks';
+import { useScreenSize } from '../../hooks';
 
-export const RowContext: React.Context<number> = createContext(initialSize);
+interface RowInterface {
+	rowsize: number;
+	rowwidth: SCREEN_SIZE;
+}
+const initialDimension: RowInterface = {
+	rowsize: 12,
+	rowwidth: 'lg',
+};
+
+export const RowContext: React.Context<RowInterface> = createContext(initialDimension);
 
 export const RowConsumer = RowContext.Consumer;
 
 export interface RowStyles {
-	root: StyleProp<ViewStyle>;
+	root: ViewStyle;
 }
 
 export interface RowProps extends ViewProps {
-	alignItems?: ViewStyle['alignItems'],
-	justifyContent?: ViewStyle['justifyContent'],
-	nowrap?: boolean,
-	rowSize?: number,
+	alignItems?: ViewStyle['alignItems'];
+	justifyContent?: ViewStyle['justifyContent'];
+	nowrap?: boolean;
+	rowSize?: number;
+	dimension: 'layout' | 'screen';
```

Comment:
this prop should be optional, and have a defaultProp value. 

@ammarkhan967 we would also need to add this prop in the @bluebase/components repo

Reply:
@artalat sir, can you please guide us, in which file of bluebase/component this prop need to add?

Reply:
https://github.com/BlueBaseJS/components/blob/master/src/components/Row.ts

Reply:
done sir